### PR TITLE
feature(dbt): transformations module

### PR DESCRIPTION
**Pull request by LukaDevic**
Ticket -> https://fivetran.height.app/user/luka/T-359251

### CODE REVIEW

File: dbt_project_details.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	CreatedAt     string `json:"created_at"`
+	CreatedById   string `json:"created_by_id"`
+	PublicKey     string `json:"public_key"`
+	DbtVersion    string `json:"dbt_version"`
+	GitRemoteUrl  string `json:"git_remote_url"`
+	GitBranch     string `json:"git_branch"`
+	DefaultSchema string `json:"default_schema"`
+	FolderPath    string `json:"folder_path"`
+	TargetName    string `json:"target_name"`
+}
```
Comment (beevital) [inline dbt_project_details.go:28]: this could be replaced with common structure

File: dbt_project_details.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+type DbtProjectDetailsResponse struct {
```
Comment (beevital) [inline dbt_project_details.go:30]: Response is the same for all three cases, so maybe we could use the same struct?

File: dbt_transformation_modify.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	DbtModelId      string    `json:"dbt_model_id`
+	OutputModelName string    `json:"output_model_name`
+	DbtProjectId    string    `json:"dbt_project_id"`
+	LastRun         time.Time `json:"last_run"`
+	NextRun         time.Time `json:"next_run"`
+	Status          string    `json:"status"`
+	RunTests        *bool     `json:"run_tests"`
+	ConnectorIds    []string  `json:"connector_ids"`
+	ModelIds        []string  `json:"model_ids"`
+}
```
Comment (beevital) [inline dbt_transformation_modify.go:41]: Same here, we could use common struct for all responses

**Review by beevital — COMMENTED** [review]
I think we could refactor data types to use common ResponseData structures for project and transformation entities.

also we need to define examples for each use-case: it's goo to have it, and use it for manual resting and debugging on live api (or testing api instance that run locally)

File: dbt_project_create.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+	return nil
+}
+
+func (s *DbtProjectCreateService) Do(ctx context.Context) (DbtProjectCreateResponse, error) {
+	var response DbtProjectCreateResponse
+
+	err := s.do(ctx, s.request(), &response)
+
+	return response, err
+}
```
[the remark this answers is not in the document]
Comment (LukaDevic) [reply dbt_project_create.go]: Done. :)

File: dbt_project_modify.go [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+	return nil
+}
+
+func (s *DbtProjectModifyService) Do(ctx context.Context) (DbtProjectModifyResponse, error) {
+	var response DbtProjectModifyResponse
+
+	err := s.do(ctx, s.request(), &response)
+
+	return response, err
+}
```
[the remark this answers is not in the document]
Comment (LukaDevic) [reply dbt_project_modify.go]: Done :)

File: dbt_project_create.go [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+	GitBranch     *string `json:"git_branch,omitempty"`
+	DefaultSchema *string `json:"default_schema,omitempty"`
+	FolderPath    *string `json:"folder_path,omitempty"`
+	TargetName    *string `json:"target_name,omitempty"`
+	Threads       *int    `json:"threads,omitempty"`
+}
+
+type dbtProjectCreateRequest struct {
+	dbtProjectCreateRequestBase
+}
```
[the remark this answers is not in the document]
Comment (LukaDevic) [reply dbt_project_create.go]: Done. :)

File: dbt_project_details.go (hunk 1 above)
Comment (LukaDevic) [reply dbt_project_details.go]: done. :)

File: dbt_project_details.go (hunk 2 above)
Comment (LukaDevic) [reply dbt_project_details.go:16]: done. :)

File: dbt_transformation_modify.go (hunk 3 above)
Comment (LukaDevic) [reply dbt_transformation_modify.go:41]: done. :)

**Comment by LukaDevic** [comment]
Feature blocked. Closing PR.

**Pull request closed without merging**